ctp2_code/ui/aui_ctp2/c3_listitem.cpp

```diff
@@ -30,7 +30,7 @@ c3_ListItem::~c3_ListItem()
 		delete m_childList->GetNext( position );
 }
 
-sint32 c3_ListItem::Compare(c3_ListItem *item2, uint32 column)
+sint32 c3_ListItem::Compare(c3_ListItem *item2, sint32 column)
```

Comment:
Can you explain why we should compare against -2? Is that not a bug in another place which we should fix in stead of making the column unsigned? Although it is indeed strange that the code contains a test on smaller than 0 on an unsigned variable.

I checked the callers and they use singed integers so I think this is a good improvement. Thanks.

Reply:
I think the use of a negative number was meant to show that we are not sorting on a column but instead doing a special type of sort - however this is only implemented in loadsavewindow.cpp, where k_LOADSAVE_AUTOSORT_COL is defined as -2. The change for this file and header was mistakenly left out of this branch when cherry-picking the commits.

This doesn't seem to be used anywhere, so perhaps a better fix might be to remove the code from loadsavewindow and change all the callers to use unsigned int. This would also mean redundant code that checks whether column is < 0 (which can never be true for an unsigned int) could be removed from many of the compare functions.
The current implementation causes a debug compile error for me on Fedora 32 linux (and presumably other distributions with the same version of gcc), as the compiler detects that -2 is outside the defined range for an unsigned int.